addons/reloadlaunchers/functions/fnc_reloadLauncher.sqf

```diff
@@ -18,12 +20,27 @@
  * Public: No
  */
 
-params ["_unit","_target","_weapon","_magazine"];
-TRACE_4("params",_unit,_target,_weapon,_magazine);
+params ["_unit", "_target", "_weapon", "_magazine", "_ammo"];
+TRACE_5("params",_unit,_target,_weapon,_magazine,_ammo);
 
-_target selectWeapon _weapon;
+private _checkSelectedWeapon = isPlayer _target || {!isNull (_target getVariable ["bis_fnc_moduleRemoteControl_owner", objNull])};
 
-if (currentWeapon _target != _weapon) exitWith {};
-if (currentMagazine _target != "") exitWith {};
+// AI don't select launchers with selectWeapon/switchWeapon
+if (_checkSelectedWeapon) then {
+    _target selectWeapon _weapon;
```

Comment:
you can reload a launcher that isn't actively selected
so it's a bit weird if someone reloads a launcher on your back and if forces you to select it suddenly

Reply:
I'm aware of that behaviour, but I kept it in as it was like that before, as I wasn't quite sure how to handle it. Should it be changed so it doesn't select the weapon?
As it is now, the launcher only needs to be in the secondary slot, but not actually equipped. I could make it so that `FUNC(canLoad)` checks if the launcher is selected.

Reply:
I think it's ok to allow reload just if the weapon exists, and it shouldn't matter if selected

Reply:
@PabstMirror I have just changed the code.
[72bf07f](https://github.com/acemod/ACE3/pull/9335/commits/72bf07f163b9b7a221d9761bffcbc1eac00b62b9) makes it so it checks if the launcher can be reloaded both before the progressbar has started and after the progressbar has finished. If it's possible, it tries to remove the magazine.
If it's not possible to load the launcher or the magazine couldn't be removed, it stops and notifies the unit that the reloading has failed.
However, if everything is valid, it will reload the launcher and notifies the unit that the reloading has succeeded.